test/e2e/specs/site-editor/navigation.spec.js

```diff
@@ -106,6 +106,25 @@ test.describe( 'Site editor navigation', () => {
 		// We should have our editor canvas button back
 		await expect( editorCanvasButton ).toBeVisible();
 	} );
+
+	test( 'Should show 404 page when navigating to non-existent template', async ( {
+		admin,
+		page,
+	} ) => {
+		await admin.visitSiteEditor();
+
+		// Navigate to a non-existent template.
+		await page.goto( '/wp-admin/site-editor.php?p=%2Ftemplate-foo-bar' );
```

Comment:
```suggestion
		await admin.visitAdminPage( 'site-editor.php', 'p=/template-foo-bar' );
```

Let's use the utility.